components/script/dom/performance.rs

```diff
@@ -56,11 +56,12 @@ const INVALID_ENTRY_NAMES: &'static [&'static str] = &[
 #[derive(JSTraceable, MallocSizeOf)]
 pub struct PerformanceEntryList {
     entries: DOMPerformanceEntryList,
+    maxSize:usize,
```

Comment:
@ferjm Thanks for the help!
Just a query
a) cant we implement resource_timing_secondary_entries as DomRefCell<PerformanceEntryList> instead of DomRefCell<Vec<DomRoot<PerformanceEntry>>>?
b) Also If we  are implementing it as DomRefCell<Vec<DomRoot<PerformanceEntry>>> , can you guide me on how to initialise it in the new_inherited function? :sweat_smile:  Thanks again :smile:

Reply:
> a) cant we implement resource_timing_secondary_entries as
DomRefCell<PerformanceEntryList> instead of DomRefCell<Vec<DomRoot<PerformanceEntry>>>?

I don't think we need the full power of a PerfomanceEntryList in this case, a Vec should be sufficient for our needs.

> b) Also If we are implementing it as DomRefCell<Vec<DomRoot>> , can you guide me on how to initialise it in the new_inherited function?

```rust
resource_timing_secondary_entries: DomRefCell::new(Vec::new()),
```